src/key.cpp

```diff
@@ -321,14 +325,12 @@ bool CKey::Derive(CKey& keyChild, ChainCode &ccChild, unsigned int nChild, const
         assert(pubkey.size() == CPubKey::COMPRESSED_SIZE);
         BIP32Hash(cc, nChild, *pubkey.begin(), pubkey.begin()+1, vout.data());
     } else {
-        assert(size() == 32);
+        assert(size() == KEY_SIZE);
         BIP32Hash(cc, nChild, 0, begin(), vout.data());
     }
-    memcpy(ccChild.begin(), vout.data()+32, 32);
-    memcpy((unsigned char*)keyChild.begin(), begin(), 32);
+    memcpy(ccChild.begin(), vout.data() + KEY_SIZE, 32);
+    keyChild.Set(begin(), begin() + KEY_SIZE, true);
     bool ret = secp256k1_ec_seckey_tweak_add(secp256k1_context_sign, (unsigned char*)keyChild.begin(), vout.data());
-    keyChild.fCompressed = true;
-    keyChild.fValid = ret;
```

Comment:
Not `if (!ret) ClearShrink(keyChild.keydata); keyChild.fValid = ret;` ? I guess the `[[nodiscard]]` is good enough though.